### ISSUE TITLE
Skip APIC detection on Uhyve

### ISSUE THREAD

**Issue by jounathaen** [hermit-os/kernel#1352]
I'm not 100% sure about this one, but AFAIK, Uhyve does not provide an APIC or IO-APIC. But if the `acpi` feature is active, the kernel unsuccessfully probes the memory to find one. I think, we can skip the search for an APIC if we are on Uhyve and save a few CPU cycles.

**Comment by stlankes** [comment]
I would assume that we need at least a local APIC to program the timer interrupt.

**Comment by jounathaen** [comment]
Yes. Something like this is happening. But the probing can probably be skipped.
